src/support/z_sites_dispatcher.erl

```diff
@@ -587,7 +587,8 @@ find_dispatch_fallback() ->
 first_site_match([], _DispHost, _ReqData) ->
     no_host_match;
 first_site_match([Site|Sites], DispHost, ReqData) ->
-    case catch z_notifier:first(DispHost, z_context:new(Site)) of
+    NewSiteContext = z_context:new(Site),
+    case catch z_notifier:first(DispHost, NewSiteContext#context{wm_reqdata=ReqData}) of
```

Comment:
The `z_context:new/1` call should be inside the catch expression.
You should use `z_context:set_reqdata(ReqData, z_context:new(Site)))` here.